serverlessworkflow/modules/ROOT/pages/data-index/data-index-quarkus-extension.adoc

```diff
@@ -0,0 +1,116 @@
+:kogito_sw_timeouts_showcase_embedded_example_url: {kogito_sw_examples_url}/serverless-workflow-timeouts-showcase-embedded
+:kogito_sw_timeouts_showcase_embedded_example_application_properties_url: {kogito_sw_timeouts_showcase_embedded_example_url}/src/main/resources/application.properties
+:infinispan_url: https://infinispan.org/
+:mongo_url: https://www.mongodb.com/
+:postgresql_url: https://www.postgresql.org/
+
+
+This document describes how you add the Data Index features to your runtime service. You simply need to add the Data Index extension to the runtime and
+all the audit data will be collected and stored in the database, and the graphQL endpoint will be available to execute queries and perform management operation over the workflows instances.
+
+The example described in this document is based on the link:{kogito_sw_timeouts_showcase_embedded_example_url}[`serverless-workflow-timeouts_showcase_embedded`] example application.
+
+[[data-index-ext-overview]]
+== Overview
+
+The Data Index service has been designed to store and manage audit data from different workflows that may be hosted in different runtimes.
+Communication with the service is through events that contain the audit data and the service is responsible for storing them and exposing a GRAPHQL endpoint to allow queries and maintenance operations on the different instances of Workflows.
+
+Sometimes, to avoid deploying the service separately, it could be interesting to have the audit storage functionality and the query possibility embedded in the same application.
+For this purpose, we have the Quarkus Data Index extension that can be plugged to any workflow application and incorporates the Data Index functionality into the same application with no need of an external Data Index Service.
+These extensions can be found distributed as Kogito addons: `kogito-addons-quarkus-data-index`.
+
+[NOTE]
+====
+The Data Index extensions are provided as addons for each kind of supported persistence relying on the link:{quarkus_guides_base_url}/writing-extensions[Quarkus extensions] mechanism.
+====
+
+Once one of these `kogito-addons-quarkus-data-index` addons is added to a Workflow application, that service incorporates the functionality to store all the audit data and also incorporates the GraphQL endpoint to perform queries and management operations.
+
+In the same way as the Data Index Service, there is a specific addon for each type of persistence you want to work with. Currently, we can find Data Index addons for: link:{postgresql_url}[PostgreSQL]
+link:{infinispan_url}[Infinispan] and  link:{mongo_url}[MongoDB]
+
+[IMPORTANT]
+====
+The `kogito-addons-quarkus-data-index` addon distribution added to the application should match with the application kind of persistence, regarding the audit data will be stored in the same database.
+The addon will share the data-source used by the application where it is added, and it will create separated audit tables for that purpose.
```

Comment:
```suggestion
The addon will share the data source used by the application where it is added, and it will create separate audit tables for that purpose.
```